src/overlays/actors/ovl_Bg_Po_Syokudai/z_bg_po_syokudai.c

```diff
@@ -21,11 +46,128 @@ const ActorInit Bg_Po_Syokudai_InitVars = {
     (ActorFunc)BgPoSyokudai_Update,
     (ActorFunc)BgPoSyokudai_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Syokudai/BgPoSyokudai_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Syokudai/BgPoSyokudai_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_060003A0[]; // Display List
+extern Gfx D_0404D4E0[]; // Display List
+
+void BgPoSyokudai_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    void* pad;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    this->unk_14C = (THIS->actor.params >> 0x8) & 0xFF;
```

Comment:
can't use `this` here?
also `0x8` can be just `8`

Reply:
Unfortunately, using `this` will create regalloc issues, where `v0` is used instead of a temp:

<pre><font color="#C4A000">24:    lh      </font><font color="#75507B">t6</font>,0x1c(<font color="#06989A">s0</font>)                        <font color="#C4A000">r 24:    lh      </font><font color="#75507B">v0</font>,0x1c(<font color="#06989A">s0</font>)                      
<font color="#CC0000">28:    lh      t9,0x1c(s0)</font>                        <font color="#CC0000">&lt;     </font>                                            
<font color="#C4A000">2c:    li      </font><font color="#4E9A06">t1</font>,0xff                            <font color="#C4A000">r 28:    li      </font><font color="#4E9A06">t9</font>,0xff                          
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 2c:    sb      t9,0xae(s0)</font>                      
<font color="#C4A000">30:    sra     </font><font color="#CC0000">t8</font>,<font color="#75507B">t6</font>,0x8                          <font color="#C4A000">r 30:    sra     </font><font color="#CC0000">t7</font>,<font color="#75507B">v0</font>,0x8                        
<font color="#C4A000">34:    andi    </font><font color="#FCE94F">t0</font>,<font color="#AD7FA8">t9</font>,0x3f                         <font color="#C4A000">r 34:    andi    </font><font color="#FCE94F">t8</font>,<font color="#75507B">v0</font>,0x3f                       
<font color="#C4A000">38:    sb      </font><font color="#CC0000">t8</font>,0x14c(<font color="#06989A">s0</font>)                       <font color="#C4A000">r 38:    sb      </font><font color="#CC0000">t7</font>,0x14c(<font color="#06989A">s0</font>)                     
<font color="#C4A000">3c:    sh      </font><font color="#FCE94F">t0</font>,0x1c(<font color="#06989A">s0</font>)                        <font color="#C4A000">r 3c:    sh      </font><font color="#FCE94F">t8</font>,0x1c(<font color="#06989A">s0</font>)                      
<font color="#CC0000">40:    sb      t1,0xae(s0)</font>                        <font color="#CC0000">&lt;     </font>                                            
44:    addiu   a2,s0,0x154                          40:    addiu   a2,s0,0x154                      
</pre>